### DISCUSSION TITLE
Create Testing checklist before merging a PR

### DISCUSSION THREAD

**Discussion by arunazhakesan** [eclipse-sw360/sw360#2171] [General]
At present, we don't do a thorough check on the impact of a new feature/change/fix on all the existing features that are working. This is breaking things intermittently and demands additional effort in fixing it (E.g. Moderation feature not working for comp & releases now). This cannot happen this way and we need to find a quick solution here. I propose that we prioritize this tasks asap.

discuss in today's (Nov 8th) meeting

**Comment by KoukiHama** [comment]
Test scenarios are available.
https://eclipse.dev/sw360/docs/development/testcases/

However, there are three challenges.

1, No one checks whether these test scenarios are not sufficient or validated.
2, It has not been determined when the tests are being carried out.
3, Tests are not sufficiently automated.

There are several possible ways to automate testing, but do we need to prepare for it with both old and new GUIs?

**Comment by KoukiHama** [comment]
> At present, we don't do a thorough check on the impact of a new feature/change/fix on all the existing features that are working. This is breaking things intermittently and demands additional effort in fixing it (E.g. Moderation feature not working for comp & releases now). This cannot happen this way and we need to find a quick solution here. I propose that we prioritize this tasks asap.
> 
> discuss in today's (Nov 8th) meeting


> E.g. Moderation feature not working for comp & releases now). 
This issue.
https://github.com/eclipse-sw360/sw360/issues/2179

**Comment by tienlee** [comment]
I agree.
We should initiate a test checklist for every PR.
The checklist can be based on [test scenarios](https://eclipse.dev/sw360/docs/development/testcases/) (should we improve these scenarios?), including basic functions (View, Create, Update, Delete, Import/Export) for each main tab (project, component, license, moderation request).

Once the checklist is stable, we can use an E2E testing framework ([Cypress](https://www.cypress.io/)?) to automate each case in the checklist.
Now, we're rolling out a new frontend. I am concerned whether we should automate for both GUIs or just for the new UI?

**Comment by heliocastro** [comment]
The interface, new UI. Is useless loose time with older one.
But the rest can be done right now.
